src/FilterPanel.tsx

```diff
@@ -0,0 +1,100 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+import { Todo } from './types/Todo';
+
+interface FilterPanelProps {
+  setFilterMode: React.Dispatch<React.SetStateAction<string>>,
+  filteredTodos: Todo[];
+  setFilteredTodos: React.Dispatch<React.SetStateAction<Todo[]>>
+}
+
+export const FilterPanel: React.FC<FilterPanelProps> = ({
+  setFilterMode, filteredTodos, setFilteredTodos,
+}) => {
+  const [activeButton, setActiveButton] = useState<string>('All');
+
+  const hasCompletedTasks = filteredTodos.some((todo) => todo.completed);
+  const activeTodoCounter = filteredTodos.reduce((counter, todo) => {
+    if (!todo.completed) {
+      // eslint-disable-next-line no-param-reassign
```

Comment:
Don't disable eslint

Reply:
haven't found a way to write this without the error

Reply:
`reduce` method should return the next value of accumulator on every iteration. So you can write the next way:
```
return counter + 1;
``` 
But don't mutate accumulator directly